tests/profiler/core/test_profiler_config_parser.py

```diff
@@ -389,32 +407,48 @@ def test_update_step_profiler_config_parser(
     assert profiler_config_parser.config.detailed_profiling_config.start_step == 5
     assert profiler_config_parser.config.detailed_profiling_config.num_steps == 2
 
-    # check that reloading the config when it hasn't changed won't change the config fields.
     profiler_config_parser.load_config()
+
+    # check that the config wasn't loaded into memory again.
+    last_accessed_time = get_last_accessed_time(step_profiler_config_parser_path)
+    assert first_accessed_time == last_accessed_time
+
+    # check that reloading the config when it hasn't changed won't change the config fields.
     assert profiler_config_parser.config.detailed_profiling_config.start_step == 5
     assert profiler_config_parser.config.detailed_profiling_config.num_steps == 2
 
-    # check that reloading the config when it has changed will update the config fields.
-    monkeypatch.setenv("SMPROFILER_CONFIG_PATH", new_step_profiler_config_parser_path)
+    shutil.copy(new_step_profiler_config_parser_path, step_profiler_config_parser_path)
```

Comment:
Tests are currently failing since this call has different behavior for Linux and Unix (for Linux, this copies the file metadata too).

Will need to fix this by finding a better API for copying files or writing the JSON manually.